state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2046,12 +2047,29 @@ pub mod pallet {
 		) {
 			let is_ccm_swap = maybe_ccm_metadata.is_some();
 
+			let aux_lookup_key = maybe_ccm_metadata.as_ref().and_then(|ccm| {
+				T::CcmValidityChecker::decode_unchecked(
+					ccm.channel_metadata.ccm_additional_data.clone(),
+					asset.into(),
+				)
+				.ok()
+				.and_then(|decoded| match decoded {
+					DecodedCcmAdditionalData::Solana(sol_ccm_data)
```

Comment:
This is also similar to the above comment on the ingress-egress pallet. It's less straightforward in the swapping pallet - given it's not instantiable. However In the same way, EgressHandler: EgressApi trait dispatches to each of the chains, hiding the details and keeping the swapping code very tidy, we could do the same here for AuxData. It should be the Any type's responsibility to do this, rather than the swapping pallet's

Reply:
I've refactored some logic into the `AnyChainCcmAuxDataLookupKey`. I'm not sure if this is good enough but I really struggle to make this more generic.